examples/utils/example-utils/src/assert.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * A browser friendly assert library.
+ *
+ * @param condition - The condition that should be true, if the condition is false an error will be thrown.
+ * Only use this API when `false` indicates a logic error in the problem and thus a bug that should be fixed.
+ * @param message - The message to include in the error when the condition does not hold.
+ *
+ * @remarks
+ * Use this instead of the node's 'assert' package, which needs a polyfill for web and has a big impact on bundle sizes.
```

Comment:
Check out this thread for more context: https://teams.microsoft.com/l/message/19:50292e8934024fc19d6ca2080dd7681e@thread.skype/1698081212555?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=9ce27575-2f82-4689-abdb-bcff07e8063b&parentMessageId=1698081212555&teamName=Fluid%20Framework&channelName=Dev&createdTime=1698081212555

Reply:
I have ported this to use a library thats reasonable for real apps to use. Its missing a "fail" variant (which is the one thats more annoying to inline since throw expressions are still missing from typescript, so you have to `?? (() => throw new Error())()` instead of `?? fail()`, but generally works ok.